mypy/semanal.py

```diff
@@ -477,6 +516,74 @@ def adjust_public_exports(self) -> None:
                 else:
                     g.module_public = False
 
+    def inherit_signature(self, cls_info: TypeInfo, func: FuncDef) -> None:
+        """Copy type annotations from parent definition, if we can find it."""
+        parent, base_1 = None, None
+        base_0 = cls_info.bases[0]
+        for base_1 in cls_info.mro[1:]:
+            if func.name() in base_1.names:
+                parent = base_1.names[func.name()].node
+                if isinstance(parent, Decorator):
+                    parent = parent.func
+                break
+        if not parent:
+            return None
+        assert base_1 is not None
+        assert isinstance(parent, FuncDef), "got %s" % type(parent)
+        pnb_args, cnb_args = len(parent.arg_kinds), len(func.arg_kinds)
+        if cnb_args < pnb_args:
+            self.fail("Signature of \"%s\" incompatible with supertype \"%s\"" %
```

Comment:
All the message generating logic should live `messages.py`. Also I think we should emit a more specific error message, explaining that we can't copy the signature, and mentioning what went wrong (not enough arguments, etc).

Also in Python you can use `'single quotes to avoid "escaping" double quotes'` (also below).